app/controllers/SchemeDetailsController.scala

```diff
@@ -97,19 +105,29 @@ class SchemeDetailsController @Inject()(appConfig: FrontendAppConfig,
                 Ok(schemeDetails(appConfig,
                   schemeName,
                   openedDate(srn.id, list, isSchemeOpen),
-                  administratorsVariations(request.psaId.id, scheme, schemeStatus),
+                  administratorsVariations(request.psaId.id, result._1, schemeStatus),
                   srn.id,
-                  isSchemeOpen
+                  isSchemeOpen,
+                  isLocked
                 ))
               }
             }
           }
         } else {
           Future.successful(NotFound(errorHandler.notFoundTemplate))
         }
-      }
     }
 
+
+  private def withSchemeAndLock(srn: SchemeReferenceNumber)(implicit request: AuthenticatedRequest[AnyContent]) = {
+    for{
+      _ <- userAnswersCacheConnector.removeAll(request.externalId)
+      scheme <- schemeDetailsConnector.getSchemeDetailsVariations(request.psaId.id, "srn", srn)
```

Comment:
Also, might be better to define this method as protected on a new parent trait so that it can be used in any other controllers in the future that need access to the lock

Reply:
This is only place in manage where we are using locks all other screen are in pension-scheme-frontend  where we already have reusable feature.